builtin/providers/librato/resource_librato_metric_test.go

```diff
@@ -0,0 +1,222 @@
+package librato
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/henrikhodne/go-librato/librato"
+)
+
+func TestAccLibratoCounterMetric(t *testing.T) {
+	var metric librato.Metric
+	name := fmt.Sprintf("tftest-metric-%s", acctest.RandString(10))
+	typ := "counter"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLibratoMetricDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: counterMetricConfig(name, typ, fmt.Sprintf("A test %s metric", typ)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLibratoMetricExists("librato_metric.foobar", &metric),
+					testAccCheckLibratoMetricName(&metric, name),
+					testAccCheckLibratoMetricType(&metric, typ),
+					resource.TestCheckResourceAttr(
+						"librato_metric.foobar", "name", name),
+				),
+			},
+			{
+				PreConfig: sleep(t, 15),
```

Comment:
Yeah, we have the same issue with the Heroku Platform API; I recently had to do something similar over there to get the test suite to run consistently. I've reduced the sleep amount to 5 secs and it still seems to be working for the most part. Removing the sleep causes the suite to consistently fail with errors caused by metrics not getting cleaned up (destroyed) properly.